### PR TITLE
Issue #7144: aligned javadoc/xdoc for IllegalImport

### PR DESCRIPTION
**Pull request by pbludov**
Issue #7144 (part of #5750)

No changes in the code.

### CODE REVIEW

File: src/xdocs/config_imports.xml [hunk 1]
```diff
@@ -620,15 +620,26 @@ import android.*;
...
           check rejects all <code>sun.*</code> packages since
           programs that contain direct calls to the <code>sun.*</code> packages are <a
           href="https://www.oracle.com/technetwork/java/faq-sun-packages-142232.html">"not
           guaranteed to work on all Java-compatible platforms"</a>. To reject other packages,
-          set property <code> illegalPkgs</code> to a list of the illegal packages.
+          set property <code>illegalPkgs</code> to a list of the illegal packages.
+        </p>
+      </subsection>
+
+      <subsection name="Notes" id="IllegalImport_Notes">
```
Comment (rnveach) [inline src/xdocs/config_imports.xml]: I just happened to think about this now. We say "See Notes" section after we have already discussed the notes on properties we have not explained yet.

File: src/xdocs/config_imports.xml [hunk 2]
```diff
@@ -620,15 +620,26 @@ import android.*;
...
       <subsection name="Description" id="IllegalImport_Description">
-        <p>Since Checkstyle 3.0</p>
         <p>
           Checks for imports from a set of illegal packages. By default, the
           check rejects all <code>sun.*</code> packages since
           programs that contain direct calls to the <code>sun.*</code> packages are <a
           href="https://www.oracle.com/technetwork/java/faq-sun-packages-142232.html">"not
           guaranteed to work on all Java-compatible platforms"</a>. To reject other packages,
-          set property <code> illegalPkgs</code> to a list of the illegal packages.
+          set property <code>illegalPkgs</code> to a list of the illegal packages.
```
Comment (rnveach) [inline src/xdocs/config_imports.xml]: Should this be in notes?

File: src/xdocs/config_imports.xml (hunk 1 above)
Comment (rnveach) [reply src/xdocs/config_imports.xml]: Should we rethink the sections? Maybe we need Property Notes and Check Notes depending on the subject of the notes? Or ....

Comment (pbludov) [reply src/xdocs/config_imports.xml]: Preperty notes below  the property table sounds good for me.

File: src/xdocs/config_imports.xml [hunk 3]
```diff
@@ -620,15 +620,26 @@ import android.*;
...
+        </p>
+      </subsection>
+
+      <subsection name="Notes" id="IllegalImport_Notes">
+        <p>
+          Note that if variable <code>regexp</code> is set to <code>false</code>,
```
Comment (romani) [inline src/xdocs/config_imports.xml]: `if property`
please recheck that in this doc there is no reference to "variable" term.

File: src/xdocs/config_imports.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config_imports.xml]: be careful with section Notes, there will be multiple notes for different properties, so we might have to make direct links to required Note item. 
I still think having very long and detailed description of property is good, but we review it after xdoc-javadoc sync.

File: src/xdocs/config_imports.xml (hunk 3 above)
Comment (pbludov) [reply src/xdocs/config_imports.xml]: done

File: src/xdocs/config_imports.xml (hunk 2 above)
Comment (pbludov) [reply src/xdocs/config_imports.xml]: done

**Comment by pbludov** [comment]
image

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java [hunk 4]
```diff
@@ -77,14 +279,14 @@
     /** The compiled regular expressions for classes. */
     private final List<Pattern> illegalClassesRegexps = new ArrayList<>();
 
-    /** List of illegal packages. */
+    /** Specify packages to reject. See details section below. */
     private String[] illegalPkgs;
 
-    /** List of illegal classes. */
+    /** Specify class names to reject. See details section below. */
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java]: With such reference, javadoc become weird.
I still think that keeping all details in one place, javadoc will looks better.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java]: done. reverted to inplace version.

**Pull request merged**